hasher-matcher-actioner/hmalib/common/aws_dataclass.py

```diff
@@ -33,6 +33,11 @@ class Item(HasAWSSerialization):
 
 T = t.TypeVar("T")
 
+# dynamodb does not allow sets to be empty. Use this as the only member when
+# writing an empty set. When retrieving a set, remove this if it is the only
+# value.
+EMPTY_SET_SENTINEL_VALUE_STR = "::EMPTY_SET_SENTINEL::"
+
 
 class AWSSerializationFailure(ValueError):
```

Comment:
(Ignorable): After writing this lovely class, I discovered a module that seems to do it much better - `dacite`. I wish I could go back in tine and switch to it. All the the CLI storage is currently powered by `dacite` instead of handrolling something similar.

Reply:
You'd still need to do some handrolling of dynamodb specific things. True some things might have been simpler.